components/insight/SRC/org/openmicroscopy/shoola/agents/fsimporter/util/FileImportComponentI.java

```diff
@@ -0,0 +1,404 @@
+ */
+
+package org.openmicroscopy.shoola.agents.fsimporter.util;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.Collection;
+import java.util.List;
+
+import omero.gateway.model.DataObject;
+import omero.gateway.model.DatasetData;
+import omero.gateway.model.FileAnnotationData;
+
+import org.openmicroscopy.shoola.env.data.model.FileObject;
+import org.openmicroscopy.shoola.env.data.model.ImportableFile;
+import org.openmicroscopy.shoola.env.data.util.Status;
+import org.openmicroscopy.shoola.util.file.ImportErrorObject;
+
+/**
+ * Component hosting the file to import and displaying the status of the 
+ * import process.
+ * 
+ * @author Domink Lindner &nbsp;&nbsp;&nbsp;&nbsp; <a
+ *         href="mailto:d.lindner@dundee.ac.uk">d.lindner@dundee.ac.uk</a>
+ */
+public interface FileImportComponentI {
+
+    /** Indicates that the container is of type <code>Project</code>. */
+    public static final int PROJECT_TYPE = 0;
+    /** Indicates that the container is of type <code>Screen</code>. */
+    public static final int SCREEN_TYPE = 1;
+    /** Indicates that the container is of type <code>Dataset</code>. */
+    public static final int DATASET_TYPE = 2;
+    /** Indicates that no container specified. */
+    public static final int NO_CONTAINER = 3;
+    /** Bound property indicating to retry an upload.*/
+    public static final String RETRY_PROPERTY = "retry";
+    /** 
+     * Bound property indicating that the error to submit is selected or not.
+     */
+    public static final String SUBMIT_ERROR_PROPERTY = "submitError";
+    /** Bound property indicating to display the error.*/
+    public static final String DISPLAY_ERROR_PROPERTY = "displayError";
+    /** Bound property indicating to cancel the import.*/
+    public static final String CANCEL_IMPORT_PROPERTY = "cancelImport";
+    /** Bound property indicating to browse the node. */
+    public static final String BROWSE_PROPERTY = "browse";
+    /** Bound property indicating to increase the number of files to import. */
+    public static final String IMPORT_FILES_NUMBER_PROPERTY = "importFilesNumber";
+    /**
+     * Bound property indicating to load the content of the log file.
+     */
+    public static final String LOAD_LOGFILEPROPERTY = "loadLogfile";
+    /**
+     * Bound property indicating to retrieve the log file.
+     */
+    public static final String RETRIEVE_LOGFILEPROPERTY = "retrieveLogfile";
+    /**
+     * Bound property indicating to show the checksums,
+     */
+    public static final String CHECKSUM_DISPLAY_PROPERTY = "checksumDisplay";
+    /** The number of extra labels for images to add. */
+    public static final int MAX_THUMBNAILS = 3;
+
+    /**
+     * Returns the file hosted by this component.
+     * 
+     * @return See above.
+     */
+    public abstract FileObject getFile();
+
+    /**
+     * Returns the file hosted by this component.
```

Comment:
Maybe a clue as to why to choose this instead of the above `getFile()`?

Reply:
No idea. There's a 'file' and an 'originalFile' in the ImportableFile, where the originalFile apparently is "The file object before possible changes". No clue, what that's supposed to mean.

Reply:
Heh, understood, It probably made sense to somebody long ago.